### ISSUE TITLE
Highlights Generation in Cricket

### ISSUE THREAD

**Issue by ashwinvaswani** [SforAiDl/Playground#88]
A good starting point would be to use the intensity (pitch / something else) of audio from commentary to filter out "exciting events" and compile the clips from those events to generate highlight.

**Comment by ashwinvaswani** [comment]
Can also look at this: https://github.com/immuno121/Automatic-highlight-generation-from-Sports-Videos

**Comment by veds12** [comment]
I would like to take this up

**Comment by veds12** [comment]
The model in the link you provided does not incorporate audio information so might not be the best suitable approach. I found a paper which suggests the use of semantic concept mining and incrporates both audio and visual data:

sci-hub.tw/10.1007/s11042-009-0337-1

Can you take a look at this and see if we can use this

**Comment by Rijul1999** [comment]
Yeah we can definitely use that imo, looks like really good work. However implementing the network might be difficult if we start from that. I think we should first build something simple, like in this article https://www.analyticsvidhya.com/blog/2019/09/guide-automatic-highlight-generation-python-without-machine-learning/, and then slowly improve it to the network in the paper. Once we have something of a base which works, imo it will be easier for us to improve on it later on.

**Comment by veds12** [comment]
Ya makes sense. I'll start implementing what's given in the link

**Comment by veds12** [comment]
I have implemented this. Should I open a PR if this approach is going to be incorporated in the library?

**Comment by ashwinvaswani** [comment]
Yes put up a PR

**Comment by veds12** [comment]
Is there any need for enhancement right now or can the issue be closed?

**Comment by Rijul1999** [comment]
I think let's keep it open for now. We are aiming to add the enhancements ultimately after all.

**Comment by ashwinvaswani** [comment]
@Rijul1999 Basic implementation is done. Check on additional work in this
